src/components/landing-page-blocks/index.tsx

```diff
@@ -0,0 +1,19 @@
+import {
+	AutosizedHeadingBlock,
+	CalloutCardBlock,
+	CardGridBlock,
+	IconCardGridBlock,
+	ParagraphBlock,
+	SectionHeadingBlock,
+} from './components'
+
+const LandingPageBlocks = () => null
+
+export default Object.assign(LandingPageBlocks, {
```

Comment:
Could you share your thoughts on storing all of these in an object instead of importing the components directly? Also have you considered using dynamic import? And do you know how the code splitting will work here, if I only need the `ParagraphBlock`, will I be importing and loading all the others when I use this `LandingPageBlocks` object?

Reply:
@kendallstrautman When @thiskevinwang was working on the new `Pagination` component, I noticed he took this approach. Maybe he can give insight into any other reasons, but what I liked is that it made it possible to render the subcomponents of the main one all in the same Swingset page, which is something we presently can't do without putting them inside of a `LiveComponent`. Having to include `LiveComponent` for every single subcomponent adds a lot of unnecessary clutter to the page if you're just trying to see all the components visually.

I also like that in `ProductRootDocsPathLandingMarketingContent` (where these subcomponents are abstracted from) we can clearly see where the subcomponents "live" or "belong" and we can also see that they're closely related to each other. And then their relationship implies that they're intended to be used together.

As for the code splitting and dynamic imports, I'm not 100% sure yet how the bundle will be impacted in subsequent tasks that will consolidate components used in other landing pages. It's something I plan to keep my eye on as I work through those tasks. As for this PR the bundle size isn't impacted so I don't see this unknown as a blocker right now. In the case that one or a couple of the subcomponents needs used, it should be possible to directly import them from their folders!

Reply:
Okay cool, this is good context to know! It's definitely an interesting pattern. My guess is that we aren't seeing much bundle activity because most of these components were already being imported and used on each product view. But it would be good to dig in more with the bundle analyzer to see how this behaves before we adopt this pattern more broadly. 

Based on the fact that things did increase slightly, makes me wonder if in fact all components are always loaded when the parent component is used. I don't think this is a blocker, but if you want to try and get clarity on whats going on there as you work through the other tasks, that would help inform whether we adopt this pattern more broadly!

Reply:
@kendallstrautman we're having the same bundle convo on two separate PRs so I just want to double-check this! I'm not seeing that the bundle has changed in this PR based on the most recent from our bundle analysis action: https://github.com/hashicorp/dev-portal/pull/1560#issuecomment-1401327732.